include/picongpu/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp

```diff
@@ -298,24 +293,65 @@ namespace ionization
 
+
+                return newMacroElectrons;
+
+            }
+
+            /* Functor implementation
+             *
+             * Ionization model specific particle creation
+             *
+             * \tparam T_parentIon type of ion species that is being ionized
+             * \tparam T_childElectron type of electron species that is created
+             * \param parentIon ion instance that is ionized
+             * \param childElectron electron instance that is created
+             */
+            template<typename T_parentIon, typename T_childElectron>
+            DINLINE void operator()(T_parentIon& parentIon,T_childElectron& childElectron)
+            {
+                /* for not mixing operations::assign up with the nvidia functor assign */
+                namespace partOp = pmacc::particles::operations;
+                /* each thread sets the multiMask hard on "particle" (=1) */
+                childElectron[multiMask_] = 1u;
+                const float_X weighting = parentIon[weighting_];
+
+                /* each thread initializes a clone of the parent ion but leaving out
+                 * some attributes:
+                 * - multiMask: reading from global memory takes longer than just setting it again explicitly
+                 * - momentum: because the electron would get a higher energy because of the ion mass
+                 * - boundElectrons: because species other than ions or atoms do not have them
+                 * (gets AUTOMATICALLY deselected because electrons do not have this attribute)
+                 */
+                auto targetElectronClone = partOp::deselect<bmpl::vector2<multiMask, momentum> >(childElectron);
+
+                partOp::assign(targetElectronClone, partOp::deselect<particleId>(parentIon));
+
+                const float_X massIon = attribute::getMass(weighting,parentIon);
+                const float_X massElectron = attribute::getMass(weighting,childElectron);
+
+                const float3_X electronMomentum (parentIon[momentum_]*(massElectron/massIon));
+
+                childElectron[momentum_] = electronMomentum;
+
+                /* conservation of momentum
+                 * \todo add conservation of mass */
+                parentIon[momentum_] -= electronMomentum;
+
                 /** ionization of the ion by reducing the number of bound electrons
                  *
-                 * optimization: only accesses global memory if the charge
-                 *               state did really change
-                 *
                  * @warning substracting a float from a float can potentially
                  *          create a negative boundElectrons number for the ion,
                  *          see #1850 for details
                  */
-                if( numNewFreeMacroElectrons > 0u )
-                    particle[ boundElectrons_ ] -= float_X( numNewFreeMacroElectrons );
+                parentIon[boundElectrons_] -= float_X(1.);
```

Comment:
offline result: I forgot that the current limitation of this implementation is, that it does not create a single, weighted e- for multi-charge-state changes in an ionization step. It instead creates as many macro-electrons as it changes the charge state, which is point for future optimization needs.